### ISSUE TITLE
Hotkey stops working after some time

### ISSUE THREAD

**Issue by NSExceptional** [Belphemur/SoundSwitch#1588]
### What happened?

I know this is really vague but after about a day of running SoundSwitch, it just stops recognizing whatever hotkey I have set to switch the inputs.

I can right click the tray icon and quit the app. Sometimes I can relaunch it, sometimes it doesn't want to start up at all. If it does start up, then the hotkey works again.

### Step to reproduce

Not sure

### Version

6.11.0

### What version of Windows are you seeing the problem on?

Windows 10

**Comment by toddydaniel** [comment]
yes! its happening with me too! the same problem!

I am experiencing issues with the application on Windows 11. The problem occurs randomly; sometimes it happens frequently, other times it takes a while to occur. The program freezes and needs to be force-closed and reopened. Sometimes it not responds to the keyboard shortcut, but if I click to open it from the taskbar, it crashes and needs to be restarted.

**Comment by DataBitz** [comment]
Seeing the same problem, have to force close and then re-open and it works again.

**Comment by vertigo220** [comment]
This has been an [ongoing issue](https://github.com/Belphemur/SoundSwitch/issues/1223) for me for a while now. I only use it a couple times a day and not every day, basically to switch between headset and speakers when starting/stopping playing a game. And 99% of the time, it does this, so I have to close it then relaunch it, and it takes _forever_ to actually relaunch. Then it works, and may or may not work to switch back once done gaming, and then doesn't work again the next day. I love this app when it works, but it's to the point it's just not worth using, since it takes longer to quit and relaunch it just to make the shortcut work when I can much more quickly simply right-click EarTrumpet (when it hasn't also become broken; as I mentioned in the other issue, these two apps are both very problematic in this sense) and change the audio device there.

**Comment by wrecklass** [comment]
Came here to say it's worse for me. I just tried SoundSwitch and found I couldn't even set HotKeys in the app. I could set something like a simple 'h' or 'g', which are unacceptable of course. However, settting something like 'Win-Alt-H' would not work. So it's uninstalled and into the recycle bin for me. I may come back if I see these issues fixed, but Audio Switch works perfectly for me now.

**Comment by smalllbuddy** [comment]
same issue, to fix this i just set up a start menu shortcut to quickly relaunch it. Just create a .bat file with the contents: 

"@echo off
taskkill /F /IM soundswitch.exe
cd "C:\Program Files\SoundSwitch"
start SoundSwitch.exe
exit"

create a shortcut to it in your start menu folder, and change the icon to soundswitch.exe icon.

** It is good to note that this issue seems to occur after extensive ram/cpu usage (in my experience) AI generation makes my entire PC freeze up but sound switch will still work, however loading up Fortnite will simply make it stop working.

**Comment by vertigo220** [comment]
Yes, that can easily be done, but even with a batch file, I can just as easily switch it myself in the same amount of time, not to mention the time it takes to realize it's not working, again. I gave up on it because I need software that works. I have enough half-working software and other things to deal with, it's just not worth it to me to be constantly dealing with this, especially since I'm no longer saving any time at that point.

It is interesting you mention Fortnite, though, because 99% of the time I'm switching audio devices when starting/stopping a gaming session, and it's usually Fortnite (or was, before I quit playing it due to too many problems with it). So maybe there's some connection between that specific game and SoundSwitch. Perhaps something the dev can look into, but I'm starting to suspect development on this app has slowed to a crawl if not stopped altogether, as it's been 7 months since the last update and 9 since they've commented on an issue.

I started writing a script to make use of AHK, SoundVolumeView, and nircmd, but AHK was not cooperating and I moved onto other things and haven't revisited it. If somebody wants to make a replacement for this, that seems like the way to go.

**Comment by NSExceptional** [comment]
You may be onto something. I haven't played Fortnite in about 2 months and all my issues with SoundSwitch have entirely vanished.
